### PR TITLE
[#2460] Reuse methods for processing Kafka headers

### PR DESCRIPTION
**Pull request by kaniyan**
This is for issue #2460

### CODE REVIEW

**Comment by kaniyan** [comment]
@b-abel As we discussed, I have raised this PR for issue #2460. Would you mind also taking a look?

**Comment by calohmn** [comment]
You could replace `new KafkaHeaderImpl` by `KafkaHeader.header` in `KafkaMessageHelper`.
Do you want to do the suggested renaming to `RecordHelper` also here as a 2nd commit?

File: clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaMessageHelper.java [hunk 1]
```diff
@@ -199,4 +202,18 @@ public static boolean isTtlElapsed(final List<KafkaHeader> headers) {
         }
     }
 
+    /**
+     * Gets the String value of a specific <em>header</em>.
+     * <p>
+     * If the headers contains multiple occurrences of the same key, the value of its first
```
Comment (calohmn) [inline clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaMessageHelper.java:208]: "contain" (also in `getHeaderValue`)

**Comment by kaniyan** [comment]
> You could replace `new KafkaHeaderImpl` by `KafkaHeader.header` in `KafkaMessageHelper`.

It was already there in the existing code and it is good to replace it with `KafkaHeader.header`

**Comment by kaniyan** [comment]
> Do you want to do the suggested renaming to `RecordHelper` also here as a 2nd commit?

FMPOV `RecordHelper` seems to be quite generic. How about `KafkaRecordHelper`?

**Comment by calohmn** [comment]
> How about KafkaRecordHelper?

I'm fine with that.

**Comment by kaniyan** [comment]
> Do you want to do the suggested renaming to `RecordHelper` also here as a 2nd commit?

@calohmn I have added a 2nd commit regarding renaming of that file and also updated in accordance with the above comments.

**Review by calohmn — APPROVED** [review]
LGTM

File: clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaRecordHelper.java [hunk 2]
```diff
@@ -199,4 +201,18 @@ public static boolean isTtlElapsed(final List<KafkaHeader> headers) {
...
+     * <p>
+     * If the headers contain multiple occurrences of the same key, the value of its first
+     * occurrence is returned.
+     *
+     * @param headers The headers to retrieve the value from.
+     * @param key The header key.
+     * @return The value or {@code null} if the headers do not contain a value of the expected type for the given key.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    public static String getStringHeaderValue(final List<KafkaHeader> headers, final String key) {
```
Comment (b-abel) [inline clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaRecordHelper.java:215]: Why does this not return an optional like the other methods here?

Comment (kaniyan) [reply clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaRecordHelper.java:215]: This method has been moved from the `RecordUtil` class (as that class has been deleted). The classes which use this utility method expect that this method returns a `String` value if present else _null_.

Comment (b-abel) [reply clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaRecordHelper.java:215]: I know that. But should not `KafkaRecordHelper` define its interface and code that uses it be adapted instead?

Comment (kaniyan) [reply clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaRecordHelper.java:215]: FMPOV it is more of a taste if to return a _String_ or an _Optional_ for a utility method. Considering that we already return _Optional_ for an already existing method in that class, I find it reasonable to have the same style. If `getStringHeaderValue(...)` is to return an _Optional_, I find it there is not much value in having that method and rather better to use the `getHeaderValue(headers, key, String.class)`.  I have made changes to use the  `getHeaderValue(headers, key, String.class)`.

**Comment by b-abel** [comment]
LGTM

**Comment by kaniyan** [comment]
The conflicts due to merge of PR #2498 has been resolved.

**Pull request merged**